### PR TITLE
docstore: API for increment modifications

### PR DESCRIPTION
**Pull request by jba**
This PR contains only the new API for an increment modification.

Implementations and tests to follow.

See #2067.

### CODE REVIEW

File: internal/docstore/docstore.go [hunk 1]
```diff
@@ -331,11 +342,30 @@ func toDriverMods(mods Mods) ([]driver.Mod, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		if inc, ok := v.(driver.IncOp); ok && !isIncNumber(inc.Amount) {
+			return nil, gcerr.Newf(gcerr.InvalidArgument, nil,
+				"Increment amount %v must be an integer or floating-point number", inc.Amount)
```
Comment (shantuo) [inline internal/docstore/docstore.go]: nit: use verb `%T`.

File: internal/docstore/memdocstore/mem.go [hunk 2]
```diff
@@ -275,6 +283,39 @@ func (c *collection) update(doc map[string]interface{}, mods []driver.Mod) error
...
 
+// Add two encoded numbers.
+// Since they're encoded, they are either int64 or float64.
+// Allow adding a float to an int, producing a float.
+// TODO(jba): see how other providers handle that.
+func add(x, y interface{}) (interface{}, error) {
+	if x == nil {
+		return y, nil
+	}
+	switch x := x.(type) {
```
Comment (shantuo) [inline internal/docstore/memdocstore/mem.go]: Is it safe to do
1. convert both to `float64` and calculate.
1. convert back if both are `int64`?

Comment (jba) [reply internal/docstore/memdocstore/mem.go]: Usually, but not always.
https://play.golang.org/p/WAv_3ZENgnx

File: internal/docstore/docstore.go (hunk 1 above)
Comment (jba) [reply internal/docstore/docstore.go]: Done.

**Pull request merged**